### DISCUSSION TITLE
assimp cannot read simple obj file

### DISCUSSION THREAD

**Discussion by pca006132** [elalish/manifold#1081] [Q&A]
I finished implementing a simple mesh dump for debugging, but apparently assimp cannot import the simple obj file properly...

For example, for the following obj file (note that vertex indices start from 1 in obj file):
```
v 5248.20561383250333165051415563 -20170.0219117954256944358348846 4599.79468714238464599475264549
v 5250.17329241749666834948584437 -20170.2593382045743055641651154 4599.52660192011535400524735451
v 5254.53032086375333165051415563 -19987.1410524204256944358348846 4484.25025354863464599475264549
v 5256.49799944874666834948584437 -19987.3784788295743055641651154 4483.98216832636535400524735451
v 5268.75395367625333165051415563 -20040.4808961704256944358348846 4635.88794886113464599475264549
v 5270.72163226124666834948584437 -20040.7183225795743055641651154 4635.61986363886535400524735451
f 4 2 1
f 1 2 6
f 1 3 4
f 3 1 5
f 2 4 6
f 6 5 1
f 4 3 5
f 5 6 4
```

Reading it with assimp via `MeshIO`, with debug information added:
```c++
  std::cout << "mNumMeshes: " << scene->mNumMeshes << std::endl;
  for (size_t i = 0; i < scene->mNumMeshes; ++i) {
    const aiMesh* mesh_i = scene->mMeshes[i];
    std::cout << "mNumVertices: " << mesh_i->mNumVertices << std::endl;
```

shows `mNumMeshes: 1` and `mNumVertices: 24`, and this causes the import to fail...

Wondering if there is a simpler way without the need of writing a simple parser for obj files.

@elalish

**Comment by elalish** [comment]
Yeah, I'm not especially thrilled with Assimp, though this is even worse. Should we consider https://github.com/3MFConsortium/lib3mf?

**Reply by pca006132** [reply]
I want to avoid extra dependencies, and ideally we should be able to dump ascii out from stdout because writing files can be annoying (name conflict, web, etc.)

I guess the issue I have is either 1) I misunderstood the obj format, 2) we should change the assimp code, because there is no way assimp can be this bad at reading obj file???

The file reads fine with things like meshlab though.

**Reply by PaddiM** [reply]
If no colors are needed, how about binary STL files?

**Reply by pca006132** [reply]
STL is definitely not a candidate here, it doesn't preserve manifoldness.

**Reply by PaddiM** [reply]
My mistake. Let me check the OBJ reader in assimp, perhaps its something simple like '/' is needed in face definitions...

**Reply by pca006132** [reply]
Yeah, I'm thinking about something like this, or it is some settings in our MeshIO code that is incompatible with the OBJ reader.

**Reply by PaddiM** [reply]
Fresh checkout of assimp - could convert your sample without problems from OBJ to STL via assimp command.

**Reply by pca006132** [reply]
Just checked, it seems that assimp requires cleanup on the imported mesh even when using the OBJ format which is already an indexed format... weird

**Reply by pca006132** [reply]
No idea how to enable reading double precision float though, there is such an option but I have no idea how to enable it, and probably easier at this point to just write my own parser...

**Reply by PaddiM** [reply]
Can you show me your call for aiImportFile? I think setting the postprocessing flags to zero can help.

**Reply by pca006132** [reply]
It probably doesn't matter too much now considering I am implementing it in https://github.com/elalish/manifold/pull/1104. Not as powerful as assimp, not spec compliant, but good enough for debug use.
